### ISSUE TITLE
do_detect() takes from 4 to 5 positional arguments but 6 were given

### ISSUE THREAD

**Issue by rmutalik** [Tianxiaomo/pytorch-YOLOv4#418]
In pytorch-YOLOv4/evaluate_on_coco.py, line 176:

**Comment by BruceDai003** [comment]
Agreed, 2 occurences of `do_detect()` need to be fixed in this `evaluate_on_coco.py` file.

**Comment by MohamedElsaeidy** [comment]
remove that 0.4 and it will work but you will get another error regarding unknown image type

**Comment by BruceDai003** [comment]
> remove that 0.4 ( confidence ) and it will work but you will get another error regarding unknown image type

I don't think you should remove the conf_thresh, instead you should remove the 80, which probably is an argument like max_num_pred, because if you check the definition of `do_detect`, you will see that there is `conf_thresh` and `nms_thresh`, but no argument like `max_num_pred`.
Regarding the datatype error, I think it should be replaced with `cv2.imread`, I have fixed this. Maybe I will do a PR later.

**Comment by rmutalik** [comment]
> I don't think you should remove the conf_thresh, instead you should remove the 80, which probably is an argument like max_num_pred, because if you check the definition of `do_detect`, you will see that there is `conf_thresh` and `nms_thresh`, but no argument like `max_num_pred`.

I think the '80' refers to the default number of classes.

**Comment by MheadHero** [comment]
> > remove that 0.4 ( confidence ) and it will work but you will get another error regarding unknown image type
> 
> I don't think you should remove the conf_thresh, instead you should remove the 80, which probably is an argument like max_num_pred, because if you check the definition of `do_detect`, you will see that there is `conf_thresh` and `nms_thresh`, but no argument like `max_num_pred`. Regarding the datatype error, I think it should be replaced with `cv2.imread`, I have fixed this. Maybe I will do a PR later.

Hi sir, can you tell replace cv2.imread with what?
